### PR TITLE
change too much gas provided message

### PR DESCRIPTION
**Pull request by sasurobert**
change too much gas provided message

### CODE REVIEW

File: process/smartContract/process.go [hunk 1]
```diff
@@ -1952,8 +1952,14 @@ func (sc *scProcessor) penalizeUserIfNeeded(
 		sc.gasHandler.SetGasPenalized(vmOutput.GasRemaining, txHash)
 	}
 
-	vmOutput.ReturnMessage += fmt.Sprintf("%s: gas needed = %d, gas remained = %d",
-		TooMuchGasProvidedMessage, gasUsed, vmOutput.GasRemaining)
+	if sc.flagCleanUpInformativeSCRs.IsSet() {
+		vmOutput.ReturnMessage += fmt.Sprintf("%s: gas needed = %d, gas remained = %d",
+			TooMuchGasProvidedMessage, gasUsed, vmOutput.GasRemaining)
+	} else {
+		vmOutput.ReturnMessage += fmt.Sprintf("%s for processing: gas needed = %d, gas remained = %d",
```
Comment (iulianpascalau) [inline process/smartContract/process.go:1959]: I would add a test here just for backwards compatibility reasons. As to not be tempted to be changed by someone else.

Comment (sasurobert) [reply process/smartContract/process.go:1959]: no-one is tempted to change a thing in the scProcessor ==))

File: process/smartContract/process.go [hunk 2]
```diff
@@ -1952,8 +1952,14 @@ func (sc *scProcessor) penalizeUserIfNeeded(
 		sc.gasHandler.SetGasPenalized(vmOutput.GasRemaining, txHash)
 	}
 
-	vmOutput.ReturnMessage += fmt.Sprintf("%s: gas needed = %d, gas remained = %d",
-		TooMuchGasProvidedMessage, gasUsed, vmOutput.GasRemaining)
+	if sc.flagCleanUpInformativeSCRs.IsSet() {
```
Comment (SebastianMarian) [inline process/smartContract/process.go:1955]: if !sc.flagCleanUpInformativeSCRs.IsSet() ?

**Review by SebastianMarian — APPROVED** [review]

File: process/smartContract/process.go (hunk 2 above)
Comment (sasurobert) [reply process/smartContract/process.go]: done

File: process/smartContract/process.go [hunk 3]
```diff
@@ -1952,12 +1952,12 @@ func (sc *scProcessor) penalizeUserIfNeeded(
...
 	}
 
-	if sc.flagCleanUpInformativeSCRs.IsSet() {
+	if !sc.flagCleanUpInformativeSCRs.IsSet() {
 		vmOutput.ReturnMessage += fmt.Sprintf("%s: gas needed = %d, gas remained = %d",
 			TooMuchGasProvidedMessage, gasUsed, vmOutput.GasRemaining)
 	} else {
-		vmOutput.ReturnMessage += fmt.Sprintf("%s for processing: gas needed = %d, gas remained = %d",
-			TooMuchGasProvidedMessage, gasProvidedForProcessing-vmOutput.GasRemaining, vmOutput.GasRemaining)
+		vmOutput.ReturnMessage += fmt.Sprintf("%s for processing: gas provided = %d, gas remained = %d",
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/smartContract/process.go:1959]: it is gas used. changed.

**Review by iulianpascalau — DISMISSED** [review]

**Review by SebastianMarian — APPROVED** [review]

File: process/smartContract/process.go (hunk 1 above)
Comment (iulianpascalau) [reply process/smartContract/process.go]: This is a good one! :1st_place_medal:

File: process/transaction/transactionCostEstimator_test.go [hunk 4]
```diff
@@ -2,6 +2,7 @@ package transaction
 
 import (
 	"errors"
+	"github.com/ElrondNetwork/elrond-go/testscommon/epochNotifier"
```
Comment (iulianpascalau) [inline process/transaction/transactionCostEstimator_test.go:5]: go imports

File: process/transaction/transactionCostEstimator.go [hunk 5]
```diff
@@ -151,17 +168,22 @@ func (tce *transactionCostEstimator) simulateTransactionCost(tx *transaction.Tra
...
-func computeGasUnitsBasedOnVMOutput(tx *transaction.Transaction, vmOutput *vmcommon.VMOutput) uint64 {
+func (tce *transactionCostEstimator) computeGasUnitsBasedOnVMOutput(tx *transaction.Transaction, vmOutput *vmcommon.VMOutput) uint64 {
 	isTooMuchGasProvided := strings.Contains(vmOutput.ReturnMessage, smartContract.TooMuchGasProvidedMessage)
 	if !isTooMuchGasProvided {
 		return tx.GasLimit - vmOutput.GasRemaining
 	}
 
-	return tx.GasLimit - extractGasRemainedFromMessage(vmOutput.ReturnMessage)
+	if tce.flagTooMuchGasV2Msg.IsSet() {
+		gasNeededForProcessing := extractGasRemainedFromMessage(vmOutput.ReturnMessage, gasUsedSlitString)
```
Comment (iulianpascalau) [inline process/transaction/transactionCostEstimator.go:178]: interesting as those new lines are not covered by tests

**Review by SebastianMarian — APPROVED** [review]

File: process/transaction/transactionCostEstimator.go (hunk 5 above)
Comment (sasurobert) [reply process/transaction/transactionCostEstimator.go:178]: TestAsyncESDTTransfer is getting here.

File: process/transaction/transactionCostEstimator_test.go (hunk 4 above)
Comment (sasurobert) [reply process/transaction/transactionCostEstimator_test.go]: done

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**